dao/src/main/java/org/thingsboard/server/dao/model/sql/AiModelSettingsEntity.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright © 2016-2025 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.server.dao.model.sql;
+
+import io.hypersistence.utils.hibernate.type.json.JsonBinaryType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import org.hibernate.annotations.Type;
+import org.hibernate.proxy.HibernateProxy;
+import org.thingsboard.server.common.data.ai.AiModelSettings;
+import org.thingsboard.server.common.data.ai.model.AiModel;
+import org.thingsboard.server.common.data.id.AiModelSettingsId;
+import org.thingsboard.server.common.data.id.TenantId;
+import org.thingsboard.server.dao.model.BaseVersionedEntity;
+import org.thingsboard.server.dao.model.ModelConstants;
+
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+
+@Getter
+@Setter
+@ToString
+@Entity
+@Table(name = ModelConstants.AI_MODEL_SETTINGS_TABLE_NAME)
+public class AiModelSettingsEntity extends BaseVersionedEntity<AiModelSettings> {
+
+    public static final Map<String, String> COLUMN_MAP = Map.of(
+            "createdTime", "created_time",
+            "provider", "(configuration -> 'providerConfig' ->> 'provider')",
+            "modelId", "(configuration -> 'modelConfig' ->> 'modelId')"
+    );
```

Comment:
Please add AiModelSettingsControllerTest and test API endpoints, and also check that this sorting works as expected